src/ruby/puppet-server-lib/puppet/server/certificate.rb

```diff
@@ -15,6 +16,7 @@ def initialize(java_cert)
     end
 
     @java_cert = java_cert
+    @oid_defns = Puppet::Server::Oids.load_custom_oid_file(Puppet[:trusted_oid_mapping_file]) + Puppet::SSL::Oids::PUPPET_OIDS
```

Comment:
@cprice404 I think that makes sense, the reason I didn't do it here is I didn't realize that was an option (I haven't done much with our JRuby stuff).

I think it would be worth refactoring all of this so we can parse the file in Clojure (I already have some code locally that's doing this) and injecting it into JRuby.